/**:
  ros__parameters:
    controller_frequency: 50.0
    initial_perception_timeout: 5.0
    wait_charge_timeout: 10.0
    dock_approach_timeout: 20.0
    undock_linear_tolerance: 0.08
    undock_angular_tolerance: 0.08
    max_retries: 3
    base_frame: "<robot_namespace>/base_link"
    fixed_frame: "<robot_namespace>/odom"
    dock_backwards: false
    dock_prestaging_tolerance: 0.5


    dock_plugins: ["panther_charging_dock"]
    panther_charging_dock:
      plugin: panther_docking::PantherChargingDock
      external_detection_timeout: 0.1
      docking_distance_threshold: 0.12
      docking_yaw_threshold: 0.15
      staging_x_offset: -0.5
      filter_coef: 0.1
      use_wibotic_info: <use_wibotic_info_param>
      wibotic_info_timeout: 1.0

    docks: ["main"]
    main:
      type: panther_charging_dock
      frame: <robot_namespace>/main_wibotic_receiver_requested_pose_link
      pose: [0.0, 0.0, 0.0] # position of the dock device (not the staging position), the front (X+) of the dock should point away from the robot

    controller:
      k_phi: 1.0
      k_delta: 2.0
      v_linear_min: 0.0
      v_linear_max: 0.1
      v_angular_max: 0.15
